File: util/txt2html.c
#include <stdio.h>

int main(int argc, char *argv[]) {
    int c;
    printf("<html>\n"
           "<head>\n"
           "  <title>%s</title>\n"
           "</head>\n"
           "<body>\n"
           "<h3>%s</h3>\n"
           "<pre>", argv[1], argv[1]);
    while ((c = getchar()) != EOF) {
        switch (c) {
            case '<': printf("&lt;"); break;
            case '>': printf("&gt;"); break;
            case '&': printf("&amp;"); break;
            default: putchar(c);
        }
    }
    printf("</pre>\n"
           "<p>\n"
           "<a href=\".\">Go to Free42 home page</a>\n"
           "</body>\n"
           "</html>\n");
    return 0;
}

File: util/txt2html.py
#!/usr/bin/python

import sys

hist = sys.stdin.read()
hist = hist.replace("&", "&amp;");
hist = hist.replace("<", "&lt;");
hist = hist.replace(">", "&gt;");

print("<html>\n"
        + "<head>\n"
        + "  <title>" + sys.argv[1] + "</title>\n"
        + "  <link rel=\"icon\" type=\"image/png\" href=\"images/free42-icon.png\">\n"
        + "</head>\n"
        + "<body>\n"
        + "<h3>" + sys.argv[1] + "</h3>\n"
        + "<pre>" + hist + "</pre>\n"
        + "<p>\n"
        + "<a href=\".\">Go to Free42 home page</a>\n"
        + "</body>\n"
        + "</html>");
